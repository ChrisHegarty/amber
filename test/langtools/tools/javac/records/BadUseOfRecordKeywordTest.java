/*
 * @test /nodynamiccopyright/
 * @summary check that record is not used as a type name
 * @compile/fail/ref=BadUseOfRecordKeywordTest.out -XDrawDiagnostics BadUseOfRecordKeywordTest.java
 */

class BadUseOfRecordKeywordTest {

    interface record {
        void m();
    }

    class record {}

    enum record {A, B}

    record record(int x);
}
