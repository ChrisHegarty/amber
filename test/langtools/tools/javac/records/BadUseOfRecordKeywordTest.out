BadUseOfRecordKeywordTest.java:9:15: compiler.err.record.not.allowed: record
BadUseOfRecordKeywordTest.java:13:11: compiler.err.record.not.allowed: record
BadUseOfRecordKeywordTest.java:15:10: compiler.err.record.not.allowed: record
BadUseOfRecordKeywordTest.java:17:12: compiler.err.record.not.allowed: record
4 errors
